Use an enum class for CSSValue Type

NathanSolomon733 opened the pull request:
#### 81dfdabf3b60c3d6ec78ad6c52c901731223d4da
<pre>
Use an enum class for CSSValue Type
<a href="https://bugs.webkit.org/show_bug.cgi?id=277660">https://bugs.webkit.org/show_bug.cgi?id=277660</a>
<a href="https://rdar.apple.com/133256102">rdar://133256102</a>

Reviewed by Tim Nguyen.

Convert the enum ClassType into an enum class.

* Source/WebCore/css/CSSAnchorValue.h:
* Source/WebCore/css/CSSAspectRatioValue.h:
* Source/WebCore/css/CSSBackgroundRepeatValue.cpp:
(WebCore::CSSBackgroundRepeatValue::CSSBackgroundRepeatValue):
* Source/WebCore/css/CSSBasicShapes.cpp:
(WebCore::CSSCircleValue::CSSCircleValue):
(WebCore::CSSEllipseValue::CSSEllipseValue):
(WebCore::CSSXywhValue::CSSXywhValue):
(WebCore::CSSRectShapeValue::CSSRectShapeValue):
(WebCore::CSSPathValue::CSSPathValue):
(WebCore::CSSPolygonValue::CSSPolygonValue):
(WebCore::CSSInsetShapeValue::CSSInsetShapeValue):
(WebCore::CSSShapeValue::CSSShapeValue):
* Source/WebCore/css/CSSBorderImageSliceValue.cpp:
(WebCore::CSSBorderImageSliceValue::CSSBorderImageSliceValue):
* Source/WebCore/css/CSSBorderImageWidthValue.cpp:
(WebCore::CSSBorderImageWidthValue::CSSBorderImageWidthValue):
* Source/WebCore/css/CSSCanvasValue.cpp:
(WebCore::CSSCanvasValue::CSSCanvasValue):
* Source/WebCore/css/CSSContentDistributionValue.cpp:
(WebCore::CSSContentDistributionValue::CSSContentDistributionValue):
* Source/WebCore/css/CSSCounterValue.cpp:
(WebCore::CSSCounterValue::CSSCounterValue):
* Source/WebCore/css/CSSCrossfadeValue.cpp:
(WebCore::CSSCrossfadeValue::CSSCrossfadeValue):
* Source/WebCore/css/CSSCursorImageValue.cpp:
(WebCore::CSSCursorImageValue::CSSCursorImageValue):
* Source/WebCore/css/CSSCustomPropertyValue.h:
* Source/WebCore/css/CSSFilterImageValue.cpp:
(WebCore::CSSFilterImageValue::CSSFilterImageValue):
* Source/WebCore/css/CSSFontFaceSrcValue.cpp:
(WebCore::CSSFontFaceSrcLocalValue::CSSFontFaceSrcLocalValue):
(WebCore::CSSFontFaceSrcResourceValue::CSSFontFaceSrcResourceValue):
* Source/WebCore/css/CSSFontFeatureValue.cpp:
(WebCore::CSSFontFeatureValue::CSSFontFeatureValue):
* Source/WebCore/css/CSSFontPaletteValuesOverrideColorsValue.h:
* Source/WebCore/css/CSSFontStyleRangeValue.h:
* Source/WebCore/css/CSSFontStyleWithAngleValue.cpp:
(WebCore::CSSFontStyleWithAngleValue::CSSFontStyleWithAngleValue):
* Source/WebCore/css/CSSFontValue.h:
* Source/WebCore/css/CSSFontVariantAlternatesValue.cpp:
(WebCore::CSSFontVariantAlternatesValue::CSSFontVariantAlternatesValue):
* Source/WebCore/css/CSSFontVariationValue.cpp:
(WebCore::CSSFontVariationValue::CSSFontVariationValue):
* Source/WebCore/css/CSSFunctionValue.cpp:
(WebCore::CSSFunctionValue::CSSFunctionValue):
* Source/WebCore/css/CSSGradientValue.h:
* Source/WebCore/css/CSSGridAutoRepeatValue.cpp:
(WebCore::CSSGridAutoRepeatValue::CSSGridAutoRepeatValue):
* Source/WebCore/css/CSSGridIntegerRepeatValue.cpp:
(WebCore::CSSGridIntegerRepeatValue::CSSGridIntegerRepeatValue):
* Source/WebCore/css/CSSGridLineNamesValue.cpp:
(WebCore::CSSGridLineNamesValue::CSSGridLineNamesValue):
* Source/WebCore/css/CSSGridTemplateAreasValue.cpp:
(WebCore::CSSGridTemplateAreasValue::CSSGridTemplateAreasValue):
* Source/WebCore/css/CSSImageSetOptionValue.cpp:
(WebCore::CSSImageSetOptionValue::CSSImageSetOptionValue):
* Source/WebCore/css/CSSImageSetValue.cpp:
(WebCore::CSSImageSetValue::CSSImageSetValue):
* Source/WebCore/css/CSSImageValue.cpp:
(WebCore::CSSImageValue::CSSImageValue):
* Source/WebCore/css/CSSLineBoxContainValue.cpp:
(WebCore::CSSLineBoxContainValue::CSSLineBoxContainValue):
* Source/WebCore/css/CSSNamedImageValue.cpp:
(WebCore::CSSNamedImageValue::CSSNamedImageValue):
* Source/WebCore/css/CSSOffsetRotateValue.h:
* Source/WebCore/css/CSSPaintImageValue.cpp:
(WebCore::CSSPaintImageValue::CSSPaintImageValue):
* Source/WebCore/css/CSSPendingSubstitutionValue.h:
* Source/WebCore/css/CSSPrimitiveValue.cpp:
(WebCore::CSSPrimitiveValue::CSSPrimitiveValue):
* Source/WebCore/css/CSSQuadValue.cpp:
(WebCore::CSSQuadValue::CSSQuadValue):
* Source/WebCore/css/CSSRayValue.h:
* Source/WebCore/css/CSSRectValue.cpp:
(WebCore::CSSRectValue::CSSRectValue):
* Source/WebCore/css/CSSReflectValue.cpp:
(WebCore::CSSReflectValue::CSSReflectValue):
* Source/WebCore/css/CSSScrollValue.h:
* Source/WebCore/css/CSSShadowValue.cpp:
(WebCore::CSSShadowValue::CSSShadowValue):
* Source/WebCore/css/CSSShapeSegmentValue.h:
(WebCore::CSSShapeSegmentValue::CSSShapeSegmentValue):
* Source/WebCore/css/CSSSubgridValue.cpp:
(WebCore::CSSSubgridValue::CSSSubgridValue):
* Source/WebCore/css/CSSTimingFunctionValue.h:
* Source/WebCore/css/CSSTransformListValue.cpp:
(WebCore::CSSTransformListValue::CSSTransformListValue):
* Source/WebCore/css/CSSUnicodeRangeValue.h:
* Source/WebCore/css/CSSValue.cpp:
(WebCore::CSSValue::visitDerived):
(WebCore::CSSValue::createDeprecatedCSSOMWrapper const):
* Source/WebCore/css/CSSValue.h:
(WebCore::CSSValue::isAnchorValue const):
(WebCore::CSSValue::isAspectRatioValue const):
(WebCore::CSSValue::isBackgroundRepeatValue const):
(WebCore::CSSValue::isBorderImageSliceValue const):
(WebCore::CSSValue::isBorderImageWidthValue const):
(WebCore::CSSValue::isCalcValue const):
(WebCore::CSSValue::isCanvasValue const):
(WebCore::CSSValue::isCircle const):
(WebCore::CSSValue::isConicGradientValue const):
(WebCore::CSSValue::isContentDistributionValue const):
(WebCore::CSSValue::isCounter const):
(WebCore::CSSValue::isCrossfadeValue const):
(WebCore::CSSValue::isCubicBezierTimingFunctionValue const):
(WebCore::CSSValue::isCursorImageValue const):
(WebCore::CSSValue::isCustomPropertyValue const):
(WebCore::CSSValue::isDeprecatedLinearGradientValue const):
(WebCore::CSSValue::isDeprecatedRadialGradientValue const):
(WebCore::CSSValue::isEllipse const):
(WebCore::CSSValue::isFilterImageValue const):
(WebCore::CSSValue::isFontFaceSrcLocalValue const):
(WebCore::CSSValue::isFontFaceSrcResourceValue const):
(WebCore::CSSValue::isFontFeatureValue const):
(WebCore::CSSValue::isFontPaletteValuesOverrideColorsValue const):
(WebCore::CSSValue::isFontStyleRangeValue const):
(WebCore::CSSValue::isFontStyleWithAngleValue const):
(WebCore::CSSValue::isFontValue const):
(WebCore::CSSValue::isFontVariantAlternatesValue const):
(WebCore::CSSValue::isFontVariationValue const):
(WebCore::CSSValue::isFunctionValue const):
(WebCore::CSSValue::isGridAutoRepeatValue const):
(WebCore::CSSValue::isGridIntegerRepeatValue const):
(WebCore::CSSValue::isGridLineNamesValue const):
(WebCore::CSSValue::isGridTemplateAreasValue const):
(WebCore::CSSValue::isImageSetOptionValue const):
(WebCore::CSSValue::isImageSetValue const):
(WebCore::CSSValue::isImageValue const):
(WebCore::CSSValue::isInsetShape const):
(WebCore::CSSValue::isLineBoxContainValue const):
(WebCore::CSSValue::isLinearGradientValue const):
(WebCore::CSSValue::isLinearTimingFunctionValue const):
(WebCore::CSSValue::isNamedImageValue const):
(WebCore::CSSValue::isOffsetRotateValue const):
(WebCore::CSSValue::isPair const):
(WebCore::CSSValue::isPath const):
(WebCore::CSSValue::isPendingSubstitutionValue const):
(WebCore::CSSValue::isPolygon const):
(WebCore::CSSValue::isPrefixedLinearGradientValue const):
(WebCore::CSSValue::isPrefixedRadialGradientValue const):
(WebCore::CSSValue::isPrimitiveValue const):
(WebCore::CSSValue::isQuad const):
(WebCore::CSSValue::isRadialGradientValue const):
(WebCore::CSSValue::isRayValue const):
(WebCore::CSSValue::isRect const):
(WebCore::CSSValue::isRectShape const):
(WebCore::CSSValue::isReflectValue const):
(WebCore::CSSValue::isScrollValue const):
(WebCore::CSSValue::isShadowValue const):
(WebCore::CSSValue::isShape const):
(WebCore::CSSValue::isShapeSegment const):
(WebCore::CSSValue::isSpringTimingFunctionValue const):
(WebCore::CSSValue::isStepsTimingFunctionValue const):
(WebCore::CSSValue::isSubgridValue const):
(WebCore::CSSValue::isTransformListValue const):
(WebCore::CSSValue::isUnicodeRangeValue const):
(WebCore::CSSValue::isValueList const):
(WebCore::CSSValue::isVariableReferenceValue const):
(WebCore::CSSValue::isViewValue const):
(WebCore::CSSValue::isXywhShape const):
(WebCore::CSSValue::isPaintImageValue const):
(WebCore::CSSValue::isGradientValue const):
(WebCore::CSSValue::isImageGeneratorValue const):
(WebCore::CSSValue::containsVector const):
* Source/WebCore/css/CSSValueList.cpp:
(WebCore::CSSValueList::CSSValueList):
* Source/WebCore/css/CSSValuePair.cpp:
(WebCore::CSSValuePair::CSSValuePair):
* Source/WebCore/css/CSSVariableReferenceValue.cpp:
(WebCore::CSSVariableReferenceValue::CSSVariableReferenceValue):
* Source/WebCore/css/CSSViewValue.h:
* Source/WebCore/css/calc/CSSCalcValue.cpp:
(WebCore::CSSCalcValue::CSSCalcValue):

Canonical link: <a href="https://commits.webkit.org/283564@main">https://commits.webkit.org/283564@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e9cb1ad50841b509c239e83281bc844f4a3b6fef

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/66085 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/45458 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/18704 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/70117 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/16695 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/68203 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/53257 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/16976 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/53043 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/11623 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/69152 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/41939 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/57213 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/33675 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/38610 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/14590 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/15571 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/60504 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/14937 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/71819 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/10040 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/14352 "Found 1 new test failure: http/wpt/mediastream/transfer-mediastreamtrack-to-worker.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/60356 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/10072 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/57275 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/60647 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/14706 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/8294 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/1934 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/41266 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/42342 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/43525 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/42086 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

zakariaridouh commented on Source/WebCore/css/CSSValue.cpp:
```diff
@@ -100,143 +100,143 @@ DEFINE_ALLOCATOR_WITH_HEAP_IDENTIFIER(CSSValue);
 template<typename Visitor> constexpr decltype(auto) CSSValue::visitDerived(Visitor&& visitor)
 {
     switch (classType()) {
```
Would be cleaner to add `using enum ClassType` here and other relevant places so you don't have to prefix all of these with `ClassType::` and can keep it as it.

https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2019/p1099r5.html

zakariaridouh commented on Source/WebCore/css/CSSValue.cpp:
```diff
@@ -382,20 +382,20 @@ void CSSValue::operator delete(CSSValue* value, std::destroying_delete_t)
 {
     switch (classType()) {
-    case ImageClass:
+    case ClassType::Image:
```
same here ;)

zakariaridouh commented on Source/WebCore/css/CSSValue.h:
```diff
@@ -71,81 +71,81 @@ class CSSValue {
-    bool isViewValue() const { return m_classType == ViewClass; }
-    bool isXywhShape() const { return m_classType == XywhShapeClass; }
-    bool isPaintImageValue() const { return m_classType == PaintImageClass; }
+    bool isAnchorValue() const { return classType() == ClassType::Anchor; }
```
Calling `classType()` for these in unnecessary, we can still just compare using `m_classType` if you change its type from an `unsigned` to `ClassType`

zakariaridouh reviewed (changes requested):
few comments :) 

I believe you can just change line 336 to:

`ClassType m_classType : ClassTypeBits; // ClassType`

And you won't need to `static_cast<>()` anymore since it's an enum class now. i.e you can change `classType()` to the following:

`constexpr ClassType classType() const { return m_classType; }`

mdubet commented on Source/WebCore/css/CSSValue.h:
```diff
@@ -217,91 +217,89 @@ class CSSValue {
-    // FIXME: Use an enum class here so we don't have to repeat "Class" in every name.
-    enum ClassType {
-        PrimitiveClass,
+    enum class ClassType : uint32_t {
```
It's not very important because we hardcode the actual field size to be 7 bits (`ClassTypeBits`), but this could be declared as a `uint8_t` if it's used somewhere else.

NathanSolomon733 replied:
Done.

NathanSolomon733 committed:
Use an enum class for CSSValue Type
https://bugs.webkit.org/show_bug.cgi?id=277660
rdar://133256102

Reviewed by Tim Nguyen.

Convert the enum ClassType into an enum class.

* Source/WebCore/css/CSSAnchorValue.h:
* Source/WebCore/css/CSSAspectRatioValue.h:
* Source/WebCore/css/CSSBackgroundRepeatValue.cpp:
(WebCore::CSSBackgroundRepeatValue::CSSBackgroundRepeatValue):
* Source/WebCore/css/CSSBasicShapes.cpp:
(WebCore::CSSCircleValue::CSSCircleValue):
(WebCore::CSSEllipseValue::CSSEllipseValue):
(WebCore::CSSXywhValue::CSSXywhValue):
(WebCore::CSSRectShapeValue::CSSRectShapeValue):
(WebCore::CSSPathValue::CSSPathValue):
(WebCore::CSSPolygonValue::CSSPolygonValue):
(WebCore::CSSInsetShapeValue::CSSInsetShapeValue):
(WebCore::CSSShapeValue::CSSShapeValue):
* Source/WebCore/css/CSSBorderImageSliceValue.cpp:
(WebCore::CSSBorderImageSliceValue::CSSBorderImageSliceValue):
* Source/WebCore/css/CSSBorderImageWidthValue.cpp:
(WebCore::CSSBorderImageWidthValue::CSSBorderImageWidthValue):
* Source/WebCore/css/CSSCanvasValue.cpp:
(WebCore::CSSCanvasValue::CSSCanvasValue):
* Source/WebCore/css/CSSContentDistributionValue.cpp:
(WebCore::CSSContentDistributionValue::CSSContentDistributionValue):
* Source/WebCore/css/CSSCounterValue.cpp:
(WebCore::CSSCounterValue::CSSCounterValue):
* Source/WebCore/css/CSSCrossfadeValue.cpp:
(WebCore::CSSCrossfadeValue::CSSCrossfadeValue):
* Source/WebCore/css/CSSCursorImageValue.cpp:
(WebCore::CSSCursorImageValue::CSSCursorImageValue):
* Source/WebCore/css/CSSCustomPropertyValue.h:
* Source/WebCore/css/CSSFilterImageValue.cpp:
(WebCore::CSSFilterImageValue::CSSFilterImageValue):
* Source/WebCore/css/CSSFontFaceSrcValue.cpp:
(WebCore::CSSFontFaceSrcLocalValue::CSSFontFaceSrcLocalValue):
(WebCore::CSSFontFaceSrcResourceValue::CSSFontFaceSrcResourceValue):
* Source/WebCore/css/CSSFontFeatureValue.cpp:
(WebCore::CSSFontFeatureValue::CSSFontFeatureValue):
* Source/WebCore/css/CSSFontPaletteValuesOverrideColorsValue.h:
* Source/WebCore/css/CSSFontStyleRangeValue.h:
* Source/WebCore/css/CSSFontStyleWithAngleValue.cpp:
(WebCore::CSSFontStyleWithAngleValue::CSSFontStyleWithAngleValue):
* Source/WebCore/css/CSSFontValue.h:
* Source/WebCore/css/CSSFontVariantAlternatesValue.cpp:
(WebCore::CSSFontVariantAlternatesValue::CSSFontVariantAlternatesValue):
* Source/WebCore/css/CSSFontVariationValue.cpp:
(WebCore::CSSFontVariationValue::CSSFontVariationValue):
* Source/WebCore/css/CSSFunctionValue.cpp:
(WebCore::CSSFunctionValue::CSSFunctionValue):
* Source/WebCore/css/CSSGradientValue.h:
* Source/WebCore/css/CSSGridAutoRepeatValue.cpp:
(WebCore::CSSGridAutoRepeatValue::CSSGridAutoRepeatValue):
* Source/WebCore/css/CSSGridIntegerRepeatValue.cpp:
(WebCore::CSSGridIntegerRepeatValue::CSSGridIntegerRepeatValue):
* Source/WebCore/css/CSSGridLineNamesValue.cpp:
(WebCore::CSSGridLineNamesValue::CSSGridLineNamesValue):
* Source/WebCore/css/CSSGridTemplateAreasValue.cpp:
(WebCore::CSSGridTemplateAreasValue::CSSGridTemplateAreasValue):
* Source/WebCore/css/CSSImageSetOptionValue.cpp:
(WebCore::CSSImageSetOptionValue::CSSImageSetOptionValue):
* Source/WebCore/css/CSSImageSetValue.cpp:
(WebCore::CSSImageSetValue::CSSImageSetValue):
* Source/WebCore/css/CSSImageValue.cpp:
(WebCore::CSSImageValue::CSSImageValue):
* Source/WebCore/css/CSSLineBoxContainValue.cpp:
(WebCore::CSSLineBoxContainValue::CSSLineBoxContainValue):
* Source/WebCore/css/CSSNamedImageValue.cpp:
(WebCore::CSSNamedImageValue::CSSNamedImageValue):
* Source/WebCore/css/CSSOffsetRotateValue.h:
* Source/WebCore/css/CSSPaintImageValue.cpp:
(WebCore::CSSPaintImageValue::CSSPaintImageValue):
* Source/WebCore/css/CSSPendingSubstitutionValue.h:
* Source/WebCore/css/CSSPrimitiveValue.cpp:
(WebCore::CSSPrimitiveValue::CSSPrimitiveValue):
* Source/WebCore/css/CSSQuadValue.cpp:
(WebCore::CSSQuadValue::CSSQuadValue):
* Source/WebCore/css/CSSRayValue.h:
* Source/WebCore/css/CSSRectValue.cpp:
(WebCore::CSSRectValue::CSSRectValue):
* Source/WebCore/css/CSSReflectValue.cpp:
(WebCore::CSSReflectValue::CSSReflectValue):
* Source/WebCore/css/CSSScrollValue.h:
* Source/WebCore/css/CSSShadowValue.cpp:
(WebCore::CSSShadowValue::CSSShadowValue):
* Source/WebCore/css/CSSShapeSegmentValue.h:
(WebCore::CSSShapeSegmentValue::CSSShapeSegmentValue):
* Source/WebCore/css/CSSSubgridValue.cpp:
(WebCore::CSSSubgridValue::CSSSubgridValue):
* Source/WebCore/css/CSSTimingFunctionValue.h:
* Source/WebCore/css/CSSTransformListValue.cpp:
(WebCore::CSSTransformListValue::CSSTransformListValue):
* Source/WebCore/css/CSSUnicodeRangeValue.h:
* Source/WebCore/css/CSSValue.cpp:
(WebCore::CSSValue::visitDerived):
(WebCore::CSSValue::createDeprecatedCSSOMWrapper const):
* Source/WebCore/css/CSSValue.h:
(WebCore::CSSValue::isAnchorValue const):
(WebCore::CSSValue::isAspectRatioValue const):
(WebCore::CSSValue::isBackgroundRepeatValue const):
(WebCore::CSSValue::isBorderImageSliceValue const):
(WebCore::CSSValue::isBorderImageWidthValue const):
(WebCore::CSSValue::isCalcValue const):
(WebCore::CSSValue::isCanvasValue const):
(WebCore::CSSValue::isCircle const):
(WebCore::CSSValue::isConicGradientValue const):
(WebCore::CSSValue::isContentDistributionValue const):
(WebCore::CSSValue::isCounter const):
(WebCore::CSSValue::isCrossfadeValue const):
(WebCore::CSSValue::isCubicBezierTimingFunctionValue const):
(WebCore::CSSValue::isCursorImageValue const):
(WebCore::CSSValue::isCustomPropertyValue const):
(WebCore::CSSValue::isDeprecatedLinearGradientValue const):
(WebCore::CSSValue::isDeprecatedRadialGradientValue const):
(WebCore::CSSValue::isEllipse const):
(WebCore::CSSValue::isFilterImageValue const):
(WebCore::CSSValue::isFontFaceSrcLocalValue const):
(WebCore::CSSValue::isFontFaceSrcResourceValue const):
(WebCore::CSSValue::isFontFeatureValue const):
(WebCore::CSSValue::isFontPaletteValuesOverrideColorsValue const):
(WebCore::CSSValue::isFontStyleRangeValue const):
(WebCore::CSSValue::isFontStyleWithAngleValue const):
(WebCore::CSSValue::isFontValue const):
(WebCore::CSSValue::isFontVariantAlternatesValue const):
(WebCore::CSSValue::isFontVariationValue const):
(WebCore::CSSValue::isFunctionValue const):
(WebCore::CSSValue::isGridAutoRepeatValue const):
(WebCore::CSSValue::isGridIntegerRepeatValue const):
(WebCore::CSSValue::isGridLineNamesValue const):
(WebCore::CSSValue::isGridTemplateAreasValue const):
(WebCore::CSSValue::isImageSetOptionValue const):
(WebCore::CSSValue::isImageSetValue const):
(WebCore::CSSValue::isImageValue const):
(WebCore::CSSValue::isInsetShape const):
(WebCore::CSSValue::isLineBoxContainValue const):
(WebCore::CSSValue::isLinearGradientValue const):
(WebCore::CSSValue::isLinearTimingFunctionValue const):
(WebCore::CSSValue::isNamedImageValue const):
(WebCore::CSSValue::isOffsetRotateValue const):
(WebCore::CSSValue::isPair const):
(WebCore::CSSValue::isPath const):
(WebCore::CSSValue::isPendingSubstitutionValue const):
(WebCore::CSSValue::isPolygon const):
(WebCore::CSSValue::isPrefixedLinearGradientValue const):
(WebCore::CSSValue::isPrefixedRadialGradientValue const):
(WebCore::CSSValue::isPrimitiveValue const):
(WebCore::CSSValue::isQuad const):
(WebCore::CSSValue::isRadialGradientValue const):
(WebCore::CSSValue::isRayValue const):
(WebCore::CSSValue::isRect const):
(WebCore::CSSValue::isRectShape const):
(WebCore::CSSValue::isReflectValue const):
(WebCore::CSSValue::isScrollValue const):
(WebCore::CSSValue::isShadowValue const):
(WebCore::CSSValue::isShape const):
(WebCore::CSSValue::isShapeSegment const):
(WebCore::CSSValue::isSpringTimingFunctionValue const):
(WebCore::CSSValue::isStepsTimingFunctionValue const):
(WebCore::CSSValue::isSubgridValue const):
(WebCore::CSSValue::isTransformListValue const):
(WebCore::CSSValue::isUnicodeRangeValue const):
(WebCore::CSSValue::isValueList const):
(WebCore::CSSValue::isVariableReferenceValue const):
(WebCore::CSSValue::isViewValue const):
(WebCore::CSSValue::isXywhShape const):
(WebCore::CSSValue::isPaintImageValue const):
(WebCore::CSSValue::isGradientValue const):
(WebCore::CSSValue::isImageGeneratorValue const):
(WebCore::CSSValue::containsVector const):
* Source/WebCore/css/CSSValueList.cpp:
(WebCore::CSSValueList::CSSValueList):
* Source/WebCore/css/CSSValuePair.cpp:
(WebCore::CSSValuePair::CSSValuePair):
* Source/WebCore/css/CSSVariableReferenceValue.cpp:
(WebCore::CSSVariableReferenceValue::CSSVariableReferenceValue):
* Source/WebCore/css/CSSViewValue.h:
* Source/WebCore/css/calc/CSSCalcValue.cpp:
(WebCore::CSSCalcValue::CSSCalcValue):

Canonical link: https://commits.webkit.org/283564@main
